woocommerce/woocommerce-android
# Edit product categories

Part of: #11061

This PR relies on https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2974 that need to be merged before. 

### Description

Adds new functionality to edit existing product categories from the product detail screen. 
There are some UI improvements in the upcoming PR for dealing with the loading states and success messages. So need to worry about those in this PR. 

### Testing instructions

1. Open the product detail screen
2. Select the category section
3. Make sure you have existing categories. If not, add a few of them.
4. Tap on any of the existing categories. Do not tap on the checkbox, tap on the list item
5. This will open the category "detail" 
6. Edit the category values
7. Click "Done"
8. Notice the category is updated successfully.
